compute_tools/src/installed_extensions.rs

```diff
@@ -0,0 +1,212 @@
+use compute_api::responses::{InstalledExtension, InstalledExtenstions};
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::fmt;
+use url::Url;
+
+use anyhow::{anyhow, Result};
+use postgres::{Client, NoTls};
+use semver::Version;
+use tokio::task;
+use tracing::{debug, info};
+
+/// We don't reuse get_existing_dbs() just for code clarity
+/// and to make database listing query here more explicit.
+///
+/// Limit the number of databases to 500 to avoid excessive load.
+fn list_dbs(client: &mut Client) -> Result<Vec<String>> {
+    // `pg_database.datconnlimit = -2` means that the database is in the
+    // invalid state
+    let databases = client
+        .query(
+            "SELECT datname FROM pg_database
+                WHERE datallowconn
+                AND datconnlimit <> - 2
+                LIMIT 500",
+            &[],
+        )?
+        .iter()
+        .map(|row| {
+            let db: String = row.get("datname");
+            db
+        })
+        .collect();
+
+    Ok(databases)
+}
+
+/// Connect to every database (see list_dbs above) and get the list of installed extensions.
+/// Same extension can be installed in multiple databases with different versions,
+/// we only keep the highest and lowest version across all databases.
+pub async fn get_installed_extensions(connstr: Url) -> Result<InstalledExtenstions> {
+    let mut connstr = connstr.clone();
+
+    task::spawn_blocking(move || {
+        let mut client = Client::connect(connstr.as_str(), NoTls)?;
+        let databases: Vec<String> = list_dbs(&mut client)?;
```

Comment:
Should we set the `search_path` here for safety?

Reply:
This comment still applies.

I guess this is a more general issue though; we use the same connection string for many queries. We should make sure to set search_path in all of them..

watch_compute_activity() also sets `application_name`. I think that's a good idea for all these internal queries. But again I guess that doesn't need to be part of this PR; there are many other existing queries in compute_ctl that don't do that.